src/shark/utilityconsole/commands/rkz/FindWeapons.java

```diff
@@ -0,0 +1,505 @@
+                break;
+
+            default:
+                Console.showMessage("Unsupported criteria encountered in constructParameterCriteria! Received " + criteria);
+        }
+        ParameterCriterion.CriteriaParameter cp = new ParameterCriterion.CriteriaParameter(criteria, data);
+        retVal = new ParameterCriterion(cp, criteriaQuantity);
+
+        return retVal;
+    }
+
+    public boolean containsShipSizes(String fullInputString) {
+        boolean retVal = false;
+        retVal = fullInputString.toLowerCase().contains("fighter")
+                || fullInputString.toLowerCase().contains("frigate")
+                || fullInputString.toLowerCase().contains("destroyer")
+                || fullInputString.toLowerCase().contains("cruiser")
+                || fullInputString.toLowerCase().contains("capital") || fullInputString.toLowerCase().contains("capital-ship") || fullInputString.toLowerCase().contains("capital_ship");
+
+        return retVal;
+    }
+
+    public String replaceShipSizesWithOrdinals(String fullInputString) {
+        // We don't really care about the position of the shipsize word,
+        // just replace all of them in one go, it'll hit what it hits, and call it done.
+        return fullInputString.toLowerCase()
+                .replaceAll("fighter", String.valueOf(ShipAPI.HullSize.FIGHTER.ordinal()))
+                .replaceAll("frigate", String.valueOf(ShipAPI.HullSize.FRIGATE.ordinal()))
+                .replaceAll("destroyer", String.valueOf(ShipAPI.HullSize.DESTROYER.ordinal()))
+                .replaceAll("cruiser", String.valueOf(ShipAPI.HullSize.CRUISER.ordinal()))
+                .replaceAll("capital", String.valueOf(ShipAPI.HullSize.CAPITAL_SHIP.ordinal()))
+                .replaceAll("capital-ship", String.valueOf(ShipAPI.HullSize.CAPITAL_SHIP.ordinal()))
+                .replaceAll("capital_ship", String.valueOf(ShipAPI.HullSize.CAPITAL_SHIP.ordinal()));
+    }
+
+    public String replaceShipSizeWithExpression(String fullInputString) {
+        // While we generally don't care about the position of the "just the shipsize word",
+        // this time we want to replace it with an expression that limits to just that particular size
+        return fullInputString.toLowerCase()
+                .replaceAll("fighter", "size = " + ShipAPI.HullSize.FIGHTER.ordinal())
+                .replaceAll("frigate", "size = " + ShipAPI.HullSize.FRIGATE.ordinal())
+                .replaceAll("destroyer", "size = " + ShipAPI.HullSize.DESTROYER.ordinal())
+                .replaceAll("cruiser", "size = " + ShipAPI.HullSize.CRUISER.ordinal())
+                .replaceAll("capital", "size = " + ShipAPI.HullSize.CAPITAL_SHIP.ordinal())
+                .replaceAll("capital-ship", "size = " + ShipAPI.HullSize.CAPITAL_SHIP.ordinal())
+                .replaceAll("capital_ship", "size = " + ShipAPI.HullSize.CAPITAL_SHIP.ordinal());
+    }
```

Comment:
to tie-in with the comment above - `"type = " + WeaponAPI.WeaponType.SOMETHING` but not ordinal. or maybe ordinal, who knows, figure it out later.

Reply:
ordinal worked out fine. done. resolving. needs re-reviewing anyway.